### PR TITLE
value's mempool cleanup

### PR DESCRIPTION
**Pull request by GuyAv46**
**Describe the changes in the pull request**

Removing unnecessary dereferences, allocations, and definitions

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/value.c [hunk 1]
```diff
@@ -24,50 +24,29 @@ static size_t RSValue_NumToString(double dd, char *buf) {
   }
 }
 
-typedef struct {
-  mempool_t *values;
-  mempool_t *fieldmaps;
```
Comment (raz-mon) [inline src/value.c:29]: Any idea why we had this in the first place?

**Review by raz-mon — APPROVED** [review]
Nice 💯

**Comment by DvirDukhan** [comment]
@GuyAv46 backport?

**Pull request merged**